rapids-cmake/cpm/init.cmake

```diff
@@ -34,11 +35,24 @@ The CPM module will be downloaded based on the state of :cmake:variable:`CPM_SOU
 same download of CPM. If those variables aren't set the file will be cached
 in the build tree of the calling project
 
-.. versionadded:: v24.04.00
-  As part of establishing rapids-cmake CPM the :cmake:command:`rapids_cpm_init` command
-  will call :cmake:command:`rapids_cpm_generate_pinned_versions` to automatically generate
-  `<CMAKE_BINARY_DIR>/rapids-cmake/pinned_versions.json` which will contain all the exact git SHAs
-  used to build cpm dependencies.
+.. versionadded:: v24.06.00
+  ```
+  CUSTOM_DEFAULT_VERSION_FILE
+  ```
+  This is an advanced option that allows projects to specify a custom default `versions.json` file that will
+  be used instead of the one packaged inside rapids-cmake. Since this allows you to specify a new default
+  `versions.json` it must contain information for all the built-in rapids-cmake packages ( cccl, fmt, rmm, etc )
+  that are used by your project.
+
+  Using a built-in rapids-cmake package without an explicit entry in the custom default version file is undefined behavior.
+
+  If multiple calls to `rapids_cpm_init` occur with different default version files being used,
+  each version file will be loaded. If a project is listed in multiple default version files, the first
+  file values will be used, and all later calls for that packaged will be ignored.  This "first to record, wins"
+  approach is used to match FetchContent, and allows parent projects to override child
+  projects.
+
+  The provided json file must follow the `versions.json` format, which is :ref:`documented here<cpm_version_format>`.
```

Comment:
```suggestion
  This is an advanced option that allows projects to specify a custom default ``versions.json`` file that will
  be used instead of the one packaged inside rapids-cmake. Since this allows you to specify a new default
  ``versions.json`` it must contain information for all the built-in rapids-cmake packages ( cccl, fmt, rmm, etc )
  that are used by your project.

  Using a built-in rapids-cmake package without an explicit entry in the custom default version file is undefined behavior.

  If multiple calls to ``rapids_cpm_init`` occur with different default version files being used,
  each version file will be loaded. If a project is listed in multiple default version files, the first
  file values will be used, and all later calls for that packaged will be ignored.  This "first to record, wins"
  approach is used to match ``FetchContent``, and allows parent projects to override child
  projects.

  The provided json file must follow the ``versions.json`` format, which is :ref:`documented here<cpm_version_format>`.
```